libs/functional/CMakeLists.txt

```diff
@@ -46,6 +46,7 @@ set(functional_headers
     hpx/functional/traits/is_bind_expression.hpp
     hpx/functional/traits/is_callable.hpp
     hpx/functional/traits/is_placeholder.hpp
+    hpx/type_traits.hpp
```

Comment:
This doesn't need to live here forever. For now it only contains `result_of` so it's enough to keep it in the `functional` module.